distributed/proxy/proxy_html.py

```diff
@@ -0,0 +1,124 @@
+from urllib import parse
+
+from tornado import web
+
+from notebook.utils import url_path_join
+from jupyter_server_proxy.handlers import ProxyHandler
+
+import tornado.ioloop
+import tornado.web
+from bokeh.server.server import BaseServer
+from tornado.httpserver import HTTPServer
+
+
+def get_host():
+    return "127.0.0.1"
+
+
+class GlobalProxyHandler(ProxyHandler):
+    """
+    A tornado request handler that proxies HTTP and websockets
+    from a port to any valid endpoint'.
+    """
+    def initialize(self, server=None, extra=None):
+        self.scheduler = server
+        self.extra = extra or {}
+
+    async def http_get(self, port, proxied_path):
+        # route here first
+        # incoming URI /proxy/{proxy}/port
+        host = self.get_argument("host", None)
+        if not host:
+            host = "127.0.0.1"
+        self.host = host
+        worker = '%s:%s' % (self.host, str(port))
+        if not check_worker_bokeh_exits(self.scheduler, worker):
+            async def _noop():
+                return None
```

Comment:
Debug code?

Reply:
We need this here to ensure we only proxy to valid worker dashboards and nothing else

Reply:
The function `_noop` doesn't seem to be used anywhere

Reply:
Oh!  Sorry, I thought you were referring to the function in general.  Yes, I'll take out `_noop`

Reply:
We can have the worker send back its dashboard address if that would help.